### PR TITLE
Upgrade to Polkadot v0.9.40

### PR DESCRIPTION
**Pull request by jasl**
https://github.com/paritytech/substrate/compare/polkadot-v0.9.39...polkadot-v0.9.40

https://github.com/paritytech/cumulus/compare/polkadot-v0.9.39...polkadot-v0.9.40

https://github.com/Phala-Network/phala-blockchain/pull/1221

In short: 0.9.40 is a minor upgrade, no visible change in this release

- wPHA bug fixing
- introduce inDEX pallet
- Fix democracy incorrect configuration that enables the functionality of Child Bounties
- Khala: Will do missing migrations, just like Phala v0.1.22
- tokenomics parameters migration (budget limit)

### CODE REVIEW

File: node/src/chain_spec/phala.rs [hunk 1]
```diff
@@ -234,8 +235,14 @@ fn genesis(
...
         phragmen_election: Default::default(),
+        phala_registry: registry_override.unwrap_or(phala_parachain_runtime::PhalaRegistryConfig {
+            workers: Vec::new(),
+            gatekeepers: Vec::new(),
+            benchmark_duration: 50,
+        }),
+        phala_computation: Default::default(),
         polkadot_xcm: phala_parachain_runtime::PolkadotXcmConfig {
-            safe_xcm_version: Some(3),
+            safe_xcm_version: Some(2),
```
Comment (h4x3rotab) [inline node/src/chain_spec/phala.rs:245]: @tolak Note that the version will not change in this upgrade. This line of the code only changes the genesis storage in a new chain.

**Review by tolak — APPROVED** [review]
SubBridge part LTGM

**Comment by jasl** [comment]
We only lack disable Phala compute functionalities, but it's OK to send a separate PR, so I'm going to merge this

File: runtime/phala/src/lib.rs [hunk 2]
```diff
@@ -277,8 +283,30 @@ construct_runtime! {
         XTransfer: xtransfer::{Pallet, Call, Storage, Event<T>} = 82,
         AssetsRegistry: assets_registry::{Pallet, Call, Storage, Event<T>} = 83,
 
+        // Phala
+        PhalaMq: pallet_mq::{Pallet, Call, Storage} = 85,
+        PhalaRegistry: pallet_registry::{Pallet, Call, Event<T>, Storage, Config<T>} = 86,
+        PhalaComputation: pallet_computation::{Pallet, Call, Event<T>, Storage, Config} = 87,
+        PhalaStakePool: pallet_stake_pool::{Pallet, Event<T>, Storage} = 88,
```
Comment (h4x3rotab) [inline runtime/phala/src/lib.rs:290]: Should we still keep stake pool?

**Pull request merged**